main.go

```diff
@@ -187,15 +215,44 @@ func mailHandler(peer smtpd.Peer, env smtpd.Envelope) error {
 		env.Data,
 	)
 	if err != nil {
-		log.Printf("delivery failed: %v\n", err)
-		return smtpd.Error{Code: 554, Message: "Forwarding failed"}
-	}
+		var smtpError smtpd.Error
+
+		switch err.(type) {
+		case *textproto.Error:
```

Comment:
Good catch! In which kind of error situation does SendMail return this?

Reply:
This was taken straight from #13. But in my testing I have seen:

```
ERRO[2021-03-13T02:47:27-05:00] delivery failed                               err_code=530 err_msg="5.7.0 Authentication Required. Learn more at\n5.7.0  https://support.google.com/mail/?p=WantAuthError h75sm6304163qke.80 - gsmtp" from=jreinhart@jrr-vaio host="smtp.gmail.com:587" peer=127.0.0.1 to="[jonathon.reinhart@gmail.com]" uuid=ada212ce-ca9d-4a2b-923a-830ce6f19e37
```

Reply:
@dannykopping do you know in which situation SendMail returns this error? I can check myself later but need a bigger screen.